### PR TITLE
chore: refactor server core

### PR DESCRIPTION
**Pull request by zllkjc**
# PR Details

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/runtime/src/runtime-context.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
 import { createContext } from 'react';
-import { BaseSSRServerContext } from '@modern-js/types';
 import { createLoaderManager } from './loader/loaderManager';
 import { runtime } from './plugin';
 
 export interface RuntimeContext {
   loaderManager: ReturnType<typeof createLoaderManager>;
   runner: ReturnType<typeof runtime.init>;
-  ssrContext?: BaseSSRServerContext;
```
Comment (leeight) [inline packages/runtime/src/runtime-context.ts:9]: 这里为啥把 any 改回去了

File: packages/server/core/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
+# @modern-js/server-plugin
```
Comment (leeight) [inline packages/server/core/CHANGELOG.md:1]: 名字改改？

File: monorepo.code-workspace [hunk 3]
```diff
@@ -360,6 +360,10 @@
       "name": "@modern-js/server-plugin",
       "path": "packages/server/plugin"
     },
+    {
+      "name": "@modern-js/server-core",
+      "path": "packages/server/core"
+    },
```
Comment (leeight) [inline monorepo.code-workspace:366]: @modern-js/server-core 和 @modern-js/server-plugin 的关系是啥？

Comment (zllkjc) [reply monorepo.code-workspace]: 这里改了个名字，之前名字和 runtime、cli 都没对齐（@modern-js/runtime-core @modern-js/core）。

File: packages/runtime/src/runtime-context.ts (hunk 1 above)
Comment (zllkjc) [reply packages/runtime/src/runtime-context.ts:9]: 之前改成非 any 是当时我发现没有 IDE 提示，然后改了之后有之前那个 mr 集测失败的问题。。刚想尝试解决那个问题，发现有很多地方要改，runtime 这块代码我没那么熟，得看看。之后我另起一个 mr 去解决这个东西，先不在这里花太多时间。

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**